File: apps/backend/src/EnterpriseAgentOs.Domain/features/Channels/ChannelMessage.cs
namespace EnterpriseAgentOs.Domain.Features.Channels;

/// <summary>
/// Value object for outbound channel messages. Serialized to JSON by the gateway.
/// </summary>
public sealed record ChannelMessage(string Kind, string Content)
{
    public static ChannelMessage Text(string content) => new("text", content);
}

File: apps/backend-java/src/main/java/com/enterpriseagentos/backendjava/domain/features/agents/dtos/ChannelMessage.java
package com.enterpriseagentos.backendjava.domain.features.agents.dtos;

public final class ChannelMessage  {
    private final String kind;
    private final String content;

    public ChannelMessage(String kind, String content) {
        this.kind = kind;
        this.content = content;
    }

    public String getKind() {
        return kind;
}

    public String kind() {
        return kind;
    }

    public String getContent() {
        return content;
}

    public String content() {
        return content;
    }
}
